site/docs/releases.md

```diff
@@ -56,8 +56,46 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
   ...
 </dependencies>
 ```
+## 0.12.0 Release Notes
 
-## 0.11.1 release notes
+Apache Iceberg 0.12.0 was released on {{ DATE }}. It consists of {{ NUMCOMMITS }} commits contributed by {{ NUMCOMMITTERS }} over a {{ NUMDAYS }} day period.
+
+High-level features:
+
+* **Core**
+  * Allow Iceberg schemas to specify one or more columns as row identifers [[\#2465](https://github.com/apache/iceberg/pull/2465)]. Note that this is a prerequisite for supporting upserts in Flink.
+  * Added Jdbc [[\#1870](https://github.com/apache/iceberg/pull/1870)] and DynamoDB [[\#2688](https://github.com/apache/iceberg/pull/2688)] Iceberg catalog implementations.
+  * Added predicate pushdown for partitions and files metadata tables [[\#2358](https://github.com/apache/iceberg/pull/2358), [\#2926](https://github.com/apache/iceberg/pull/2926)]
+  * Added a new Compaction API for IcebergV2 and implemented a new compaction strategy [[\#2501](https://github.com/apache/iceberg/pull/2501), [\#2609](https://github.com/apache/iceberg/pull/2609)].
```

Comment:
This isn't really user-facing. I think the main update to mention is that there is a new, more flexible compaction action for Spark that can support different strategies, like bin packing and sorting. This also isn't really related to v2.

Reply:
Done